### PR TITLE
add skylake platform implementation

### PR DESCRIPTION
**Pull request by wanweiqiangintel**
In the skylake platform, simdjson library will fallback to haswell due to lack of AVX512-VBMI2, but this platform still can use AVX512F and so on.  In the performance test of skylake platform with AVX512(not include VBMI2), there are some benefit(about 9.7%) than AVX2 in json parse.  Should we add the implementation as supplement?
image

### CODE REVIEW

**Comment by lemire** [comment]
> In the skylake platform, simdjson library will fallback to haswell due to lack of AVX512-VBMI2, but this platform still can use AVX512F and so on. 

The issue is frequency throttling, not instruction support. See our documentation on this topic...

"Thus there should be no downclocking due to simdjson on recent processors, except when AVX-512 is allowed and detected. However, we only allow AVX-512 on recent processors (Ice Lake/Tiger Lake or better, AMD Zen 4 or better) where [little to no frequency throttling is expected](https://travisdowns.github.io/blog/2020/08/19/icl-avx512-freq.html)."

I should rephrase it to be clearer, but the idea is that we engineer the library to avoid downclocking. That's a major concern for many engineers.

**Comment by lemire** [comment]
So a skylake AVX-512 kernel would be acceptable, but I would want it *disabled by default*. It is frustrating, but we want to avoid pushing bad surprises on people. We don't want an engineer to come back to us and say that we cause unexpected frequency throttling, slowing down their application. (Note that this is not good for Intel also!!! I am sure Intel agrees that we should seek to please engineers.)

**Comment by wanweiqiangintel** [comment]
> So a skylake AVX-512 kernel would be acceptable, but I would want it _disabled by default_. It is frustrating, but we want to avoid pushing bad surprises on people. We don't want an engineer to come back to us and say that we cause unexpected frequency throttling, slowing down their application. (Note that this is not good for Intel also!!! I am sure Intel agrees that we should seek to please engineers.)

Exactly, we should avoid unexpected frequency throttling. So I close this PR. Thanks.
